language-design.md

```diff
@@ -38,16 +79,42 @@ you know a poorly designed language? What are the symptoms?
 
+> Every method should do the least surprising thing it could, given its name. If a method doesn’t do what users think it will, bugs will result. [Bloch 2006]
 
----
+Additionally it should be designed to take as much of the load off of the user as possible - for example, if a large amount of boiler plate code is required to do a certain task, that boiler plate should be pulled into the language itself and only require the minimum amount of input from the user.  "Don’t make the client do anything the library could do" [Bloch 2006]. This leads into the additional point of productivity - users should be able to use the language without fighting it, and be able to complete their tasks in a productive manner.
+
+Furthermore, in the case of a DSL, it should be usable by domain experts who have little to no knowledge of programming.
+  
+> [The] enabling of software development by users with less domain and programming expertise, or even by end-users with some domain, but virtually no
+programming expertise [Nardi 1993; Sutcliffe and Mehandjiev 2004]. [Mernik _et al._, 2005]
+
+Likewise, it should be difficult to misuse (and ideally easy to use for what it is designed as well).  
  
+> **APIs should be easy to use and hard to misuse.** It should be
+easy to do simple things; possible to do complex things; and
+impossible, or at least difficult, to do wrong things. [Block, 2006].
+
+"What is easy to do?  What is hard to do?  What is impossible?" [Prof Ben, 2015].
 
+---
+ 
+**Question**
+ 
 In what way is an API a language? 
 
 **Response**
 
+An API is a way to communicate a series of instructions to a computer. It defines a set of concepts and abstractions about objects and manipulations, to make it easier to use and understand by humans. It defines exactly what is easy to do, what is possible to do, and what cannot or shouldn't be done.
+ 
+> **APIs should be easy to use and hard to misuse.** It should be
+easy to do simple things; possible to do complex things; and
+impossible, or at least difficult, to do wrong things. [Block, 2006].
+
+It is often difficult for the users to directly modify an API. However, it still needs to be able to be used flexibly and extensibly, in situations and for purposes that were not foreseen when it was developed. Things that can be problematic for a language can also be problematic for an API, from the need of documentation to violating the principle of least surprise. The main difference is that in an API, your choices of syntax and power are more limited than the general case. However, as it tends more and more towards a true internal DSL, that boundary can also become blurrier.
```

Comment:
Unfortunately, I agree with your premise. APIs are developed by the program designer and cannot be altered by users since the source is inaccessible. However, I think users can define another abstraction that let's them do API calls in a more fluent way (like wrapper methods).